diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -169,6 +170,19 @@ fun ASTNode.isBeginByNewline() =
                     (it.treePrev.elementType == IMPORT_LIST && it.treePrev.isLeaf() && it.treePrev.treePrev.isLeaf())
         } ?: false
 
+/**
+ * Checks if there is a newline before this element or before comment before. See [isBeginByNewline] for motivation on parents check.
+ */
+fun ASTNode.isBeginNewLineWithComment() =
+        isBeginByNewline() ||
+                if (this.treePrev?.treePrev?.isPartOfComment() ?: false) {
```

Comment:
Maybe, something like `siblings(forward = false).takeWhile { !it.textContains('\n') }.all { it.isWhiteSpace() || it.isPartOfComment() }`? This will allow not to hard-code checking only `treePrev` or `teePrev.treePrev`.